Adding h2 in-memory database

mcruzdev opened the pull request:
Hello @salaboy,

After some days I created the PR from issue #2 

I added the h2 in-memory database and I changed the application.properties file

After I think to implement a repository/DAO layer to remove the in-memory storage like a monolith.

mcruzdev commented:
@salaboy I will extract **ConfererenceSiteUtilController** and **ConferenceSiteController** to separate the resposability from class.

salaboy commented:
@mcruzdev sorry for the miss communication, but yes.. please add the repository/dao layer here.. so someone is using H2, you can just keep commiting to your branch to add commits to this PR

salaboy reviewed (commented):
@mcruzdev awesome stuff.. can I ask you for one more thing to be added here? can we add some tests for the common endpoints? those are: newProposal, deleteProposal and decision. 
It will be great if we had some tests to check the outputs for those methods.

salaboy commented on src/test/java/com/salaboy/conferences/c4p/controller/C4PControllerIT.java:
```diff
@@ -0,0 +1,99 @@
+        var responseSpec = createProposalRequest();
+
+        // assert
+        responseSpec.expectBody(Proposal.class);
```
@mcruzdev can you assert that the proposal has an ID as well? As the backend should create the ID

salaboy commented on src/test/java/com/salaboy/conferences/c4p/controller/C4PControllerIT.java:
```diff
@@ -0,0 +1,99 @@
+        var requestProposal =
+                new Proposal("Title", "Description", "Author", "email@email.com");
+
+        Mockito.when(proposalRepository.save(requestProposal)).thenReturn(requestProposal);
```
@mcruzdev do you really need to mock the DB with h2?

salaboy commented on src/test/java/com/salaboy/conferences/c4p/controller/C4PControllerIT.java:
```diff
@@ -0,0 +1,99 @@
+                .expectStatus().isOk();
+
+        getAllProposals().expectBodyList(Proposal.class).hasSize(0);
+    }
```
@mcruzdev we should have a test here for the decision endpoint.. that is the most tricky one.. can you please add that one?

salaboy reviewed (commented):
@mcruzdev I've added two small comments and one big about testing the decision endpoint. This is looking good.. one thing that you will need to do for major open-source projects, is to squash all the commits and describe in detail what the commit is adding. Feel free to try this out here..

mcruzdev commented:
In my last commit, I implemented tests for decision route.
Removed direct mutability from Proposal model, and implemented two methods to be the domain intention clean
Now the test code check the proposal's id

salaboy reviewed (approved):
@mcruzdev great stuff... I will merge now.. and I will be updating the build system
